### PR TITLE
implement SMTP email notification, fixes #1

### LINKED ISSUE #1: smtp email sending is missing
the outlook-based code is not very useful for users who don't use outlook, obviously.

but python has email-creation and -sending code in the stdlib, it is just not very easy to use...

but i had some code from another FOSS project (mostly written by me), which i can contribute - PR coming soon.

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: impf_termin_tester/utils/send_mail.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    me = '"Joe Doe-Sender" <joe@example.org>'
+    them = '"Jane Doe-Receiver" <jane@example.org>'
+    # note: if sending to many people, respect their privacy and put them ALL into BCC,
+    #       so you do not disclose all their email addresses to all of them!
+    #       of course, you need to put also something into TO - just use same address as FROM for that.
+    ok, msg = send_mail(mail_from=me, to=[me, ], bcc=[them, them, them, ],
+                        subject="Test send_mail", text="Test!",
+                        mail_server="mail.example.org:587", login_user="username", login_password="password")
+    print("ok: %r, msg: %r" % (ok, msg))
+
```
Comment (andreasdoerr) [inline impf_termin_tester/utils/send_mail.py:131]: Do you wanna leave this in here as some kind of demo code? feels a bit detached from the project itself.

**Review by andreasdoerr — APPROVED** [review]
Thanks for providing the e-mail notification. Nice to have that channel!

File: impf_termin_tester/utils/send_mail.py (hunk 1 above)
Comment (ThomasWaldmann) [reply impf_termin_tester/utils/send_mail.py:131]: Yes, this is testing code. It won't disturb anything when importing this as a library, due to the if in line 120.

But it serves as an example, as testing code, and as useful hint how to send mass mails, so please keep it.

File: impf_termin_tester/notifiers/mail.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        super().__init__(name="SMTP e-mail client")
+        self.target_email = target_email
+        self.mail_server = mail_server
+        self.login_user = login_user
+        self.login_password = login_password
+
+    def initialize(self):
+        pass
+
+    def _send_notification(self, result):
```
Comment (ThomasWaldmann) [inline impf_termin_tester/notifiers/mail.py:17]: updated!

Comment (andreasdoerr) [reply impf_termin_tester/notifiers/mail.py:17]: LGTM. Just checking.. can you actually merge the PR? (since i approved). Still pretty new to github :-)

Comment (ThomasWaldmann) [reply impf_termin_tester/notifiers/mail.py:17]: No, you need to merge it.

**Pull request merged**